framework/decode/vulkan_replay_consumer_base.cpp

```diff
@@ -4774,6 +4819,112 @@ VkResult VulkanReplayConsumerBase::OverrideCreateDebugUtilsMessengerEXT(
                 pMessenger->GetHandlePointer());
 }
 
+// Callback functions required to work with the compatibility library since it has no direct
+// knowledge of the VulkanResourceAllocator.
+// ------------------------
+VkResult CreateImageResource(void*                        allocator_class,
+                             const VkImageCreateInfo*     create_info,
+                             const VkAllocationCallbacks* allocation_callbacks,
+                             VkImage*                     image,
+                             uintptr_t*                   resource_id)
+{
+    gfxrecon::decode::VulkanResourceAllocator* resource_allocator =
```

Comment:
This is going to require more work because the VulkanResourceAllocator uses objects out of the decode path.  (The various `Info` structures and the HandleId as examples).  The plan is to eventually port it over, but this change is already massive as it is.